test/test.js

```diff
@@ -32,18 +32,18 @@ test('call Intl.NumberFormat correctly', (t) => {
     this.format = () => ({ locale, opts })
   }
 
-  t.is(formatNumber(0, { maxSignificant: 2 }).opts.maximumSignificantDigits, 2)
-  t.is(formatNumber(0, { maxSignificantDigits: 2 }).opts.maximumSignificantDigits, 2)
-  t.is(formatNumber(0, { maximumSignificant: 2 }).opts.maximumSignificantDigits, 2)
-  t.is(formatNumber(0, { minSignificant: 2 }).opts.minimumSignificantDigits, 2)
-  t.is(formatNumber(0, { minSignificantDigits: 2 }).opts.minimumSignificantDigits, 2)
-  t.is(formatNumber(0, { minimumSignificant: 2 }).opts.minimumSignificantDigits, 2)
-  t.is(formatNumber(0, { maxInteger: 2 }).opts.maximumIntegerDigits, 2)
-
-  t.deepEqual(formatNumber(0).locale, ['en-US'])
-  t.deepEqual(formatNumber(0).opts, {
+  t.is(formatNumber(0, { maxSignificant: 3 }).opts.maximumSignificantDigits, 3)
+  t.is(formatNumber(0, { maxSignificantDigits: 3 }).opts.maximumSignificantDigits, 3)
+  t.is(formatNumber(0, { maximumSignificant: 3 }).opts.maximumSignificantDigits, 3)
+  t.is(formatNumber(0, { minSignificant: 3 }).opts.minimumSignificantDigits, 3)
+  t.is(formatNumber(0, { minSignificantDigits: 3 }).opts.minimumSignificantDigits, 3)
+  t.is(formatNumber(0, { minimumSignificant: 3 }).opts.minimumSignificantDigits, 3)
+  t.is(formatNumber(0, { maxInteger: 3 }).opts.maximumIntegerDigits, 3)
+
+  t.deepEqual(formatNumber(0, { locale: 'ka-GE' }).locale, ['ka-GE'])
+  t.deepEqual(formatNumber(0, { locale: 'ka-GE' }).opts, {
     nanZero: true,
-    locale: 'en-US',
+    locale: 'ka-GE',
```

Comment:
Since the format function is overwritten by a "hack" we need to instantiate number formatters with different (non memoized) opts to fix the tests

Reply:
Nit u can reset cache with

memoizedFunction.cache = new _.memoize.Cache();

Reply:
Do I need to expose `getNumberFormatter` for this?

Reply:
ah right we don't have access to it here. this solution also seems fine